File: APP/src/app/interfaces/items/magic-item.ts
import { Item, ResourceReference } from '../item';

export interface MagicItemRarity {
  name: string;
}

export interface MagicItem extends Item {
  index: string;
  name: string;
  type: string;
  equipment_category: ResourceReference;
  rarity: MagicItemRarity;
  variants: ResourceReference[];
  variant: boolean;
  desc: string[];
  image?: string;
  url: string;
}

File: API/models/item_types/MagicItem.py
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field
from models.Generico import BaseSchema, ResourceReference


class MagicItemRaritySchema(BaseModel):
	name: str

	model_config = ConfigDict(extra="allow")


class MagicItemSchema(BaseSchema):
	index: Optional[str] = None
	name: str
	type: str = Field(default="magicItem")
	equipment_category: ResourceReference
	rarity: MagicItemRaritySchema
	variants: List[ResourceReference] = Field(default_factory=list)
	variant: bool = False
	desc: List[str] = Field(default_factory=list)
	image: Optional[str] = None
	url: Optional[str] = None

	model_config = ConfigDict(extra="allow")